src/main/java/com/redhat/devtools/lsp4ij/settings/ui/LanguageServerPanel.java

```diff
@@ -0,0 +1,154 @@
+package com.redhat.devtools.lsp4ij.settings.ui;
+
+import com.intellij.openapi.ui.ComboBox;
+import com.intellij.ui.PortField;
+import com.intellij.ui.components.JBCheckBox;
+import com.intellij.ui.components.JBScrollPane;
+import com.intellij.ui.components.JBTabbedPane;
+import com.intellij.ui.components.JBTextField;
+import com.intellij.ui.scale.JBUIScale;
+import com.intellij.util.ui.FormBuilder;
+import com.intellij.util.ui.components.BorderLayoutPanel;
+import com.redhat.devtools.lsp4ij.LanguageServerBundle;
+import com.redhat.devtools.lsp4ij.settings.ServerTrace;
+import org.jetbrains.annotations.NotNull;
+
+import javax.swing.*;
+import java.awt.*;
+
+public class LanguageServerPanel {
+
+    public static enum EditionMode {
+        NEW_USER_DEFINED,
+        EDIT_USER_DEFINED,
+        EDIT_EXTENSION;
+    }
+
+    private JBTextField serverName;
+    private CommandLineWidget commandLine;
+    private ServerMappingsPanel mappingsPanel;
+
+    private final PortField debugPortField = new PortField();
+    private final JBCheckBox debugSuspendCheckBox = new JBCheckBox(LanguageServerBundle.message("language.server.debug.suspend"));
+    private final ComboBox<ServerTrace> serverTraceComboBox = new ComboBox<>(new DefaultComboBoxModel<>(ServerTrace.values()));
+
+    private LanguageServerConfigurationWidget configurationWidget;
+
+    public LanguageServerPanel(FormBuilder builder, EditionMode mode) {
+        createUI(builder, mode);
+    }
+
+    private void createUI(FormBuilder builder, EditionMode mode) {
+        JBTabbedPane tabbedPane = new JBTabbedPane();
+        builder.addComponent(tabbedPane);
+
+        // Server tab
+        addServerTab(tabbedPane, mode);
+        // Mappings tab
+        addMappingsTab(tabbedPane, mode);
+        if (mode != EditionMode.EDIT_EXTENSION) {
+            // Configuration tab
+            addConfigurationTab(tabbedPane);
+        }

```

Comment:
```java
if (mode != EditionMode.EDIT_EXTENSION) {
    // Server tab
    addServerTab(tabbedPane, mode);
    // Mappings tab
    addMappingsTab(tabbedPane, mode);
    // Configuration tab
    addConfigurationTab(tabbedPane);
}
```

Reply:
Why you don't want to show mappings for extension?

Reply:
because only mappings of user defined servers currently show here.
I'm not against showing them for extension provided servers, but they would probably have to be read-only

Reply:
It was a bug that I have fixed, list of mappings are refreshed and are in read only:

![image](https://github.com/redhat-developer/lsp4ij/assets/1932211/092bee4c-020e-45b5-a308-86a0bf4ee629)